src/test/util_tests.cpp

```diff
@@ -2094,4 +2096,13 @@ BOOST_AUTO_TEST_CASE(message_verify)
         MessageVerificationResult::OK);
 }
 
+BOOST_AUTO_TEST_CASE(message_hash)
+{
+    const std::string unsigned_tx = "...";
+    const uint256 signature_hash = Hash(unsigned_tx.begin(), unsigned_tx.end());
+
+    const uint256 message_hash = MessageHash(unsigned_tx);
+    BOOST_CHECK_NE(message_hash, signature_hash);
```

Comment:
I think you should have on additional check here with the full hashed message. i.e. also hash `Bitcoin Signed Message:\n...` and compare that with `message_hash`.

Reply:
Done.

Reply:
Notice: assuming `CHashWriter hasher(SER_GETHASH, 0);`, then:
```
hasher << "ab" << "cd";
```
would produce a different result than:
```
hasher << "abcd";
```